src/Infrastructure/Persistence/PostgresUserRepository.hs

```diff
@@ -1,55 +1,63 @@
+{-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Infrastructure.Persistence.PostgresUserRepository where
 
-import Control.Arrow ((&&&))
 import Control.Monad.IO.Class (liftIO)
-import Control.Monad.Trans.Except (ExceptT (ExceptT), withExceptT)
-import Data.Bifunctor (second)
+import Control.Monad.Trans.Except (ExceptT (ExceptT), throwE, withExceptT)
 import Data.ByteString (isInfixOf)
 import Data.Text (Text)
 import Data.UUID.V4 (nextRandom)
-import Hasql.Session (CommandError (ResultError), QueryError (QueryError), ResultError (ServerError))
+import Hasql.Session (CommandError (ResultError), QueryError (QueryError), ResultError (ServerError), Session)
 import qualified Infrastructure.Database as DB
-import qualified Infrastructure.Persistence.Queries as Query (addUser, selectUserByName)
+import Infrastructure.Persistence.Queries (WrongNumberOfRows)
+import qualified Infrastructure.Persistence.Queries as Query
 import Infrastructure.Persistence.Schema (litUser, userId)
 import Infrastructure.Persistence.Serializer (serializeUser, unserializeUser)
 import Tagger.EncryptedPassword (EncryptedPassword)
 import Tagger.Id (Id (Id))
 import Tagger.User (User (User))
-import Tagger.UserRepository (SelectUserError, UserRepository (..))
+import Tagger.UserRepository (UserRepository (..))
 
 -- We want to distinguish the `QueryError` coming from the violation of the "users_name_key" unique constraints
 data UserRepositoryError
   = DuplicateUserName QueryError
+  | IncorrectNumberOfRows WrongNumberOfRows
```

Comment:
I understand that we are inside the context of Postgres, but I would try to avoid the usage of the word `Row` here, since this gives a very database flavour to the error name. We could use `Result` instead.
Also, I would use `Unexpected` instead of `Incorrect`
```suggestion
  | UnexpectedNumberOfResults WrongNumberOfResults
```